main.tf

```diff
@@ -370,6 +388,14 @@ resource "aws_route_table" "internet_gateway" {
   )
 }
 
+resource "aws_route" "internet_gateway_firewall" {
+  count = var.create_vpc && var.enable_firewall &&length(var.firewall_subnets) > 0 ? length(var.public_subnets) : 0
+
+  route_table_id         = aws_route_table.internet_gateway[0].id
+  destination_cidr_block = aws_subnet.public[count.index].cidr_block
+  vpc_endpoint_id        = tolist(aws_networkfirewall_firewall.this[0].firewall_status[0].sync_states)[count.index].attachment[0].endpoint_id
```

Comment:
does this work?

when i tried creating the firewall resource, the sync_states list is not nicely sorted by availability zone, for e.g

```
"sync_states" = toset([
        {
          "attachment" = tolist([
            {
              "endpoint_id" = "vpce-08b584fd4cafe30b4"
              "subnet_id" = "subnet-03ac99a17edc68b8a"
            },
          ])
          "availability_zone" = "ap-southeast-1c"
        },
        {
          "attachment" = tolist([
            {
              "endpoint_id" = "vpce-0baae3c68baed0a6b"
              "subnet_id" = "subnet-0dbe2bf3b5e642e82"
            },
          ])
          "availability_zone" = "ap-southeast-1b"
        },
        {
          "attachment" = tolist([
            {
              "endpoint_id" = "vpce-0c8f3d4578786d1e8"
              "subnet_id" = "subnet-06a949809ce8a4d9e"
            },
          ])
          "availability_zone" = "ap-southeast-1a"
        },
      ])
```

wheres aws_subnet.public will be created in the order of azs, 1a followed by 1b etc

So could this result in the first endpoint vpce-08b584fd4cafe30b4 using the route table that connects to public subnet in 1a?

Reply:
i use the suggestion [here](https://github.com/hashicorp/terraform-provider-aws/issues/16759#issuecomment-745536796) so it will look like
```hcl
vpc_endpoint_id = element([for ss in tolist(aws_networkfirewall_firewall.this[0].firewall_status[0].sync_states) :
  ss.attachment[0].endpoint_id if ss.attachment[0].subnet_id == aws_subnet.firewall[count.index].id], 0)```